src/Spryker/Zed/AppPayment/Business/Payment/PreOrder/PaymentPreOrder.php

```diff
@@ -0,0 +1,178 @@
+    ) {
+    }
+
+    public function confirmPreOrderPayment(
+        ConfirmPreOrderPaymentRequestTransfer $confirmPreOrderPaymentRequestTransfer
+    ): ConfirmPreOrderPaymentResponseTransfer {
+        try {
+            $confirmPreOrderPaymentRequestTransfer->setAppConfigOrFail($this->appConfigLoader->loadAppConfig($confirmPreOrderPaymentRequestTransfer->getTenantIdentifierOrFail()));
+            $confirmPreOrderPaymentRequestTransfer->setPayment($this->appPaymentRepository->getPaymentByTransactionId($confirmPreOrderPaymentRequestTransfer->getTransactionIdOrFail()));
+
+            // When the payment platform plugin does not implement the `AppPreOrderPaymentPlatformPluginInterface` we assume there is no further action from the PSP implementation
+            // needed, and we can simply update the Payment with the missing orderReference.
+            $confirmPreOrderPaymentResponseTransfer = new ConfirmPreOrderPaymentResponseTransfer();
+            $confirmPreOrderPaymentResponseTransfer
+                ->setIsSuccessful(true)
+                ->setStatus(PaymentStatus::STATUS_CAPTURED);
+
+            if ($this->appPaymentPlatformPlugin instanceof AppPaymentPlatformConfirmPreOrderPluginInterface) {
+                $confirmPreOrderPaymentResponseTransfer = $this->appPaymentPlatformPlugin->confirmPreOrderPayment($confirmPreOrderPaymentRequestTransfer);
+            }
+        } catch (Throwable $throwable) {
+            $this->getLogger()->error($throwable->getMessage(), [
+                PaymentTransfer::TENANT_IDENTIFIER => $confirmPreOrderPaymentRequestTransfer->getTenantIdentifierOrFail(),
+            ]);
+            $confirmPreOrderPaymentResponseTransfer = new ConfirmPreOrderPaymentResponseTransfer();
+            $confirmPreOrderPaymentResponseTransfer
+                ->setIsSuccessful(false)
+                ->setMessage($throwable->getMessage());
+
+            return $confirmPreOrderPaymentResponseTransfer;
+        }
+
+        /** @phpstan-var \Generated\Shared\Transfer\ConfirmPreOrderPaymentResponseTransfer */
+        return $this->getTransactionHandler()->handleTransaction(function () use ($confirmPreOrderPaymentRequestTransfer, $confirmPreOrderPaymentResponseTransfer) {
+            $this->savePayment($confirmPreOrderPaymentRequestTransfer, $confirmPreOrderPaymentResponseTransfer);
+
+            // In case of pre-order payment we may have unprocessed webhook requests persisted and we must process them here
+            $webhookInboxCriteriaTransfer = new WebhookInboxCriteriaTransfer();
+
+            // Unprocessed webhooks will be persisted by the transaction id
+            $webhookInboxCriteriaTransfer->addIdentifier($confirmPreOrderPaymentRequestTransfer->getTransactionIdOrFail());
+
+            $this->appPaymentToAppWebhookFacade->processUnprocessedWebhooks($webhookInboxCriteriaTransfer);
+
+            $this->sendMessages($confirmPreOrderPaymentRequestTransfer, $confirmPreOrderPaymentResponseTransfer);
```

Comment:
Why should it be sent to SCOS on the pre order?

- OMS events won't be processed (no order)
- PaymentUpdated won't be processed (no order)

Reply:
It's the ConfirmPreOrder payment request made from SCOS after the order was placed. We have the order on SCOS here.